src/model/cloud-runner/remote-client/caching.test.ts

```diff
@@ -0,0 +1,65 @@
+import fs from 'fs';
+import path from 'path';
+import BuildParameters from '../../build-parameters';
+import { CLI } from '../../cli/cli';
+import Input from '../../input';
+import UnityVersioning from '../../unity-versioning';
+import CloudRunner from '../cloud-runner';
+import { CloudRunnerSystem } from '../services/cloud-runner-system';
+import { Caching } from './caching';
+
+function guid() {
+  return Math.trunc((1 + Math.random()) * 0x10000)
+    .toString(16)
+    .slice(1);
+}
+function guidGenerator() {
+  return `${guid() + guid()}-${guid()}-${guid()}-${guid()}-${guid()}${guid()}${guid()}`;
+}
+describe('Cloud Runner Caching', () => {
+  it('responds', () => {});
+});
+describe('Cloud Runner Caching', () => {
+  if (process.platform === 'linux') {
+    it('Simple caching works', async () => {
+      CLI.options = {
+        versioning: 'None',
+        projectPath: 'test-project',
+        unityVersion: UnityVersioning.read('test-project'),
+        targetPlatform: 'StandaloneLinux64',
+        cacheKey: `test-case-${guidGenerator()}`,
+      };
+      Input.githubInputEnabled = false;
+      const buildParameter = await BuildParameters.create();
+      CloudRunner.buildParameters = buildParameter;
+      // create test folder
+      const testFolder = path.resolve(__dirname, CLI.options.cacheKey);
+      fs.mkdirSync(testFolder);
+      // crate cache folder
+      const cacheFolder = path.resolve(__dirname, `cache-${CLI.options.cacheKey}`);
+      fs.mkdirSync(cacheFolder);
+      // add test has file to test folders
```

Comment:
I've worked with `fs` a few times in the past and it looks like `fs.promises` is a very good and modern core library of node. Recommend using them in async context.

(see also all other occurrences)